warning: ‘QLineF::IntersectType QLineF::intersect(const QLineF&, QPointF*) const’ is deprecated

```shell
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp: In member function ‘virtual void OpenMS::Annotation1DPeakItem::draw(OpenMS::Plot1DCanvas*, QPainter&, bool)’:
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp:161:29: warning: ‘QLineF::IntersectType QLineF::intersect(const QLineF&, QPointF*) const’ is deprecated: Use intersects() instead [-Wdeprecated-declarations]
  161 |       itype = line.intersect(top, ip);
      |               ~~~~~~~~~~~~~~^~~~~~~~~
In file included from /usr/include/qt/QtGui/qmatrix.h:47,
                 from /usr/include/qt/QtGui/qbrush.h:49,
                 from /usr/include/qt/QtGui/qpen.h:45,
                 from /usr/include/qt/QtGui/QPen:1,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotations1DContainer.h:41,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/LayerDataBase.h:51,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/PlotCanvas.h:44,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/Plot1DCanvas.h:41,
                 from /home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp:36:
/usr/include/qt/QtCore/qline.h:256:19: note: declared here
  256 |     IntersectType intersect(const QLineF &l, QPointF *intersectionPoint) const;
      |                   ^~~~~~~~~
```

One more file has this
```shell
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DCaret.cpp: In member function ‘virtual void OpenMS::Annotation1DCaret::draw(OpenMS::Plot1DCanvas*, QPainter&, bool)’:
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DCaret.cpp:161:29: warning: ‘QLineF::IntersectType QLineF::intersect(const QLineF&, QPointF*) const’ is deprecated: Use intersects() instead [-Wdeprecated-declarations]
  161 |       itype = line.intersect(top, ip);
      |               ~~~~~~~~~~~~~~^~~~~~~~~
In file included from /usr/include/qt/QtGui/qmatrix.h:47,
                 from /usr/include/qt/QtGui/qtransform.h:43,
                 from /usr/include/qt/QtGui/qstatictext.h:48,
                 from /usr/include/qt/QtGui/QStaticText:1,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/ANNOTATION/Annotation1DCaret.h:40,
                 from /home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DCaret.cpp:35:
/usr/include/qt/QtCore/qline.h:256:19: note: declared here
  256 |     IntersectType intersect(const QLineF &l, QPointF *intersectionPoint) const;
      |                   ^~~~~~~~~
```

--- a/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DCaret.cpp
+++ b/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DCaret.cpp
@@ -158,7 +158,15 @@ namespace OpenMS
       bool found_intersection = false;
 
       // intersection with top
+      /*
+       * These pragmas can be removed when we are on Qt 5.14 or later
+       * QLineF::IntersectType QLineF::intersect() is deprecated at Qt 5.14
+       */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(top, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(caret_position_widget, *ip).length() < QLineF(caret_position_widget, *closest_ip).length())
       {
@@ -166,7 +174,11 @@ namespace OpenMS
         *closest_ip = *ip;
       }
       // intersection with left
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(left, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(caret_position_widget, *ip).length() < QLineF(caret_position_widget, *closest_ip).length())
       {
@@ -175,7 +187,11 @@ namespace OpenMS
       }
 
       // intersection with right
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(right, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(caret_position_widget, *ip).length() < QLineF(caret_position_widget, *closest_ip).length())
       {
@@ -184,7 +200,11 @@ namespace OpenMS
       }
 
       // intersection with bottom
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(bottom, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(caret_position_widget, *ip).length() < QLineF(caret_position_widget, *closest_ip).length())
       {

--- a/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp
+++ b/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp
@@ -158,7 +158,15 @@ namespace OpenMS
       bool found_intersection = false;
 
       // intersection with top
+      /*
+       * These pragmas can be removed when we are on Qt 5.14 or later
+       * QLineF::IntersectType QLineF::intersect() is deprecated at Qt 5.14
+       */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(top, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(peak_position_widget, *ip).length() < QLineF(peak_position_widget, *closest_ip).length())
       {
@@ -166,7 +174,11 @@ namespace OpenMS
         *closest_ip = *ip;
       }
       // intersection with left
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(left, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(peak_position_widget, *ip).length() < QLineF(peak_position_widget, *closest_ip).length())
       {
@@ -175,7 +187,11 @@ namespace OpenMS
       }
 
       // intersection with right
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(right, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(peak_position_widget, *ip).length() < QLineF(peak_position_widget, *closest_ip).length())
       {
@@ -184,7 +200,11 @@ namespace OpenMS
       }
 
       // intersection with bottom
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
       itype = line.intersect(bottom, ip);
+#pragma GCC diagnostic pop
+
       if (itype == QLineF::BoundedIntersection &&
           QLineF(peak_position_widget, *ip).length() < QLineF(peak_position_widget, *closest_ip).length())
       {

